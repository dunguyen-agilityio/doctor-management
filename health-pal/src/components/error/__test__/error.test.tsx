import { render } from '@utils-test'

// Adjust path

import ErrorState from '..'

describe('ErrorState', () => {
  it('matches snapshot with default props', () => {
    const { toJSON } = render(<ErrorState />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with custom title and message', () => {
    const { toJSON } = render(
      <ErrorState title="Network Error" message="Failed to connect to the server." />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with onRetry button', () => {
    const onRetry = jest.fn()
    const { toJSON } = render(<ErrorState onRetry={onRetry} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('matches snapshot without onRetry button', () => {
    const { toJSON } = render(<ErrorState title="Custom Error" message="Please try again later." />)
    expect(toJSON()).toMatchSnapshot()
  })
})
