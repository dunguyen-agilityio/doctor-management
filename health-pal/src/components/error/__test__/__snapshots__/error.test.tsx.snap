// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorState matches snapshot with custom title and message 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    role="alert"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "column",
        "gap": 22,
        "justifyContent": "center",
        "paddingLeft": 48,
        "paddingRight": 48,
      }
    }
  >
    <View
      style={[]}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="76"
        bbWidth="76"
        color="$red10"
        fill="none"
        focusable={false}
        height="76"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 76,
              "width": 76,
            },
          ]
        }
        vbHeight={76}
        vbWidth={76}
        width="76"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M37.9998 9.50002C22.2597 9.50002 9.49984 22.2599 9.49984 38C9.49984 53.7401 22.2597 66.5 37.9998 66.5C53.74 66.5 66.4998 53.7401 66.4998 38C66.4998 22.2599 53.74 9.50002 37.9998 9.50002ZM3.1665 38C3.1665 18.7621 18.7619 3.16669 37.9998 3.16669C57.2378 3.16669 72.8332 18.7621 72.8332 38C72.8332 57.2379 57.2378 72.8334 37.9998 72.8334C18.7619 72.8334 3.1665 57.2379 3.1665 38Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M49.739 26.2609C50.9757 27.4975 50.9757 29.5025 49.739 30.7392L30.739 49.7392C29.5023 50.9759 27.4973 50.9759 26.2607 49.7392C25.024 48.5025 25.024 46.4975 26.2607 45.2609L45.2607 26.2609C46.4973 25.0242 48.5024 25.0242 49.739 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M26.2607 26.2609C27.4973 25.0242 29.5023 25.0242 30.739 26.2609L49.739 45.2609C50.9757 46.4975 50.9757 48.5025 49.739 49.7392C48.5024 50.9759 46.4973 50.9759 45.2607 49.7392L26.2607 30.7392C25.024 29.5025 25.024 27.4975 26.2607 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        {
          "color": "#1c2a3a",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 20,
          "lineHeight": 30,
        }
      }
      suppressHighlighting={true}
    >
      Network Error
    </Text>
    <Text
      style={
        {
          "color": "#111928",
          "fontFamily": "Inter_400Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      suppressHighlighting={true}
    >
      Failed to connect to the server.
    </Text>
  </View>,
]
`;

exports[`ErrorState matches snapshot with default props 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    role="alert"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "column",
        "gap": 22,
        "justifyContent": "center",
        "paddingLeft": 48,
        "paddingRight": 48,
      }
    }
  >
    <View
      style={[]}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="76"
        bbWidth="76"
        color="$red10"
        fill="none"
        focusable={false}
        height="76"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 76,
              "width": 76,
            },
          ]
        }
        vbHeight={76}
        vbWidth={76}
        width="76"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M37.9998 9.50002C22.2597 9.50002 9.49984 22.2599 9.49984 38C9.49984 53.7401 22.2597 66.5 37.9998 66.5C53.74 66.5 66.4998 53.7401 66.4998 38C66.4998 22.2599 53.74 9.50002 37.9998 9.50002ZM3.1665 38C3.1665 18.7621 18.7619 3.16669 37.9998 3.16669C57.2378 3.16669 72.8332 18.7621 72.8332 38C72.8332 57.2379 57.2378 72.8334 37.9998 72.8334C18.7619 72.8334 3.1665 57.2379 3.1665 38Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M49.739 26.2609C50.9757 27.4975 50.9757 29.5025 49.739 30.7392L30.739 49.7392C29.5023 50.9759 27.4973 50.9759 26.2607 49.7392C25.024 48.5025 25.024 46.4975 26.2607 45.2609L45.2607 26.2609C46.4973 25.0242 48.5024 25.0242 49.739 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M26.2607 26.2609C27.4973 25.0242 29.5023 25.0242 30.739 26.2609L49.739 45.2609C50.9757 46.4975 50.9757 48.5025 49.739 49.7392C48.5024 50.9759 46.4973 50.9759 45.2607 49.7392L26.2607 30.7392C25.024 29.5025 25.024 27.4975 26.2607 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        {
          "color": "#1c2a3a",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 20,
          "lineHeight": 30,
        }
      }
      suppressHighlighting={true}
    >
      Oops!
    </Text>
    <Text
      style={
        {
          "color": "#111928",
          "fontFamily": "Inter_400Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      suppressHighlighting={true}
    >
      Something went wrong.
    </Text>
  </View>,
]
`;

exports[`ErrorState matches snapshot with onRetry button 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    role="alert"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "column",
        "gap": 22,
        "justifyContent": "center",
        "paddingLeft": 48,
        "paddingRight": 48,
      }
    }
  >
    <View
      style={[]}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="76"
        bbWidth="76"
        color="$red10"
        fill="none"
        focusable={false}
        height="76"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 76,
              "width": 76,
            },
          ]
        }
        vbHeight={76}
        vbWidth={76}
        width="76"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M37.9998 9.50002C22.2597 9.50002 9.49984 22.2599 9.49984 38C9.49984 53.7401 22.2597 66.5 37.9998 66.5C53.74 66.5 66.4998 53.7401 66.4998 38C66.4998 22.2599 53.74 9.50002 37.9998 9.50002ZM3.1665 38C3.1665 18.7621 18.7619 3.16669 37.9998 3.16669C57.2378 3.16669 72.8332 18.7621 72.8332 38C72.8332 57.2379 57.2378 72.8334 37.9998 72.8334C18.7619 72.8334 3.1665 57.2379 3.1665 38Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M49.739 26.2609C50.9757 27.4975 50.9757 29.5025 49.739 30.7392L30.739 49.7392C29.5023 50.9759 27.4973 50.9759 26.2607 49.7392C25.024 48.5025 25.024 46.4975 26.2607 45.2609L45.2607 26.2609C46.4973 25.0242 48.5024 25.0242 49.739 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M26.2607 26.2609C27.4973 25.0242 29.5023 25.0242 30.739 26.2609L49.739 45.2609C50.9757 46.4975 50.9757 48.5025 49.739 49.7392C48.5024 50.9759 46.4973 50.9759 45.2607 49.7392L26.2607 30.7392C25.024 29.5025 25.024 27.4975 26.2607 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        {
          "color": "#1c2a3a",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 20,
          "lineHeight": 30,
        }
      }
      suppressHighlighting={true}
    >
      Oops!
    </Text>
    <Text
      style={
        {
          "color": "#111928",
          "fontFamily": "Inter_400Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      suppressHighlighting={true}
    >
      Something went wrong.
    </Text>
    <View
      accessibilityHint="Retries loading"
      aria-label="Retry"
      focusVisibleStyle={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1c2a3a",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderTopWidth": 1,
          "color": "#ffffff",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "fontWeight": "500",
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 12,
          "width": "100%",
        }
      }
      testID="retry-button"
    >
      <Text
        lineBreakMode="clip"
        numberOfLines={1}
        style={
          {
            "color": "#ffffff",
            "flexGrow": 0,
            "flexShrink": 1,
            "fontFamily": "Inter_500Medium",
            "fontSize": 14,
            "lineHeight": 21,
            "userSelect": "none",
          }
        }
        suppressHighlighting={true}
      >
        Retry
      </Text>
    </View>
  </View>,
]
`;

exports[`ErrorState matches snapshot without onRetry button 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    role="alert"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "column",
        "gap": 22,
        "justifyContent": "center",
        "paddingLeft": 48,
        "paddingRight": 48,
      }
    }
  >
    <View
      style={[]}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="76"
        bbWidth="76"
        color="$red10"
        fill="none"
        focusable={false}
        height="76"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 76,
              "width": 76,
            },
          ]
        }
        vbHeight={76}
        vbWidth={76}
        width="76"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M37.9998 9.50002C22.2597 9.50002 9.49984 22.2599 9.49984 38C9.49984 53.7401 22.2597 66.5 37.9998 66.5C53.74 66.5 66.4998 53.7401 66.4998 38C66.4998 22.2599 53.74 9.50002 37.9998 9.50002ZM3.1665 38C3.1665 18.7621 18.7619 3.16669 37.9998 3.16669C57.2378 3.16669 72.8332 18.7621 72.8332 38C72.8332 57.2379 57.2378 72.8334 37.9998 72.8334C18.7619 72.8334 3.1665 57.2379 3.1665 38Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M49.739 26.2609C50.9757 27.4975 50.9757 29.5025 49.739 30.7392L30.739 49.7392C29.5023 50.9759 27.4973 50.9759 26.2607 49.7392C25.024 48.5025 25.024 46.4975 26.2607 45.2609L45.2607 26.2609C46.4973 25.0242 48.5024 25.0242 49.739 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M26.2607 26.2609C27.4973 25.0242 29.5023 25.0242 30.739 26.2609L49.739 45.2609C50.9757 46.4975 50.9757 48.5025 49.739 49.7392C48.5024 50.9759 46.4973 50.9759 45.2607 49.7392L26.2607 30.7392C25.024 29.5025 25.024 27.4975 26.2607 26.2609Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        {
          "color": "#1c2a3a",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 20,
          "lineHeight": 30,
        }
      }
      suppressHighlighting={true}
    >
      Custom Error
    </Text>
    <Text
      style={
        {
          "color": "#111928",
          "fontFamily": "Inter_400Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      suppressHighlighting={true}
    >
      Please try again later.
    </Text>
  </View>,
]
`;
